lib/src/screens/unspent_coins/unspent_coins_list_page.dart

```diff
@@ -1,30 +1,68 @@
-import 'package:cake_wallet/bitcoin_cash/bitcoin_cash.dart';
 import 'package:cake_wallet/generated/i18n.dart';
 import 'package:cake_wallet/routes.dart';
 import 'package:cake_wallet/src/screens/base_page.dart';
 import 'package:cake_wallet/src/screens/unspent_coins/widgets/unspent_coins_list_item.dart';
+import 'package:cake_wallet/src/widgets/alert_with_no_action.dart.dart';
+import 'package:cake_wallet/src/widgets/standard_checkbox.dart';
+import 'package:cake_wallet/themes/extensions/cake_text_theme.dart';
 import 'package:cake_wallet/view_model/unspent_coins/unspent_coins_list_view_model.dart';
-import 'package:cw_core/wallet_type.dart';
-import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_mobx/flutter_mobx.dart';
+import 'package:mobx/mobx.dart';
 
 class UnspentCoinsListPage extends BasePage {
   UnspentCoinsListPage({required this.unspentCoinsListViewModel});
 
   @override
   String get title => S.current.unspent_coins_title;
 
+  @override
+  Widget leading(BuildContext context) {
+    return MergeSemantics(
+      child: SizedBox(
+        height: 37,
+        width: 37,
+        child: ButtonTheme(
+          minWidth: double.minPositive,
+          child: Semantics(
+            label: S.of(context).seed_alert_back,
+            child: TextButton(
+              style: ButtonStyle(
+                overlayColor: MaterialStateColor.resolveWith((states) => Colors.transparent),
+              ),
+              onPressed: () async => await handleOnPopInvoked(context),
+              child: backButton(context),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+
   final UnspentCoinsListViewModel unspentCoinsListViewModel;
 
+  Future<void> handleOnPopInvoked(BuildContext context) async {
+    final hasChanged = unspentCoinsListViewModel.hasAdjustableFieldChanged;
+    if (unspentCoinsListViewModel.items.isEmpty || !hasChanged) {
+      Navigator.of(context).pop();
+    } else {
+      unspentCoinsListViewModel.isDisposing = true;
+      await unspentCoinsListViewModel.dispose();
+      Navigator.of(context).pop();
+      Navigator.of(context).pop();
+    }
+  }
+
   @override
-  Widget body(BuildContext context) => UnspentCoinsListForm(unspentCoinsListViewModel);
+  Widget body(BuildContext context) =>
+      UnspentCoinsListForm(unspentCoinsListViewModel, handleOnPopInvoked);
 }
 
 class UnspentCoinsListForm extends StatefulWidget {
-  UnspentCoinsListForm(this.unspentCoinsListViewModel);
+  UnspentCoinsListForm(this.unspentCoinsListViewModel, this.handleOnPopInvoked);
 
   final UnspentCoinsListViewModel unspentCoinsListViewModel;
+  final Future<void> Function(BuildContext context) handleOnPopInvoked;
```

Comment:
**minor**
can be ignored since you already use `widget.handleOnPopInvoked`

Reply:
UnspentCoinsListForm is a separate widget, so it should be there